roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python
+'''
+  Module for performing checks on an elasticseatch, fluentd, and kibana stack
+'''
+# pylint: disable=missing-docstring
+# vim: expandtab:tabstop=4:shiftwidth=4
+#
+#   Copyright 2017 Red Hat Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+import json
+import os
+import ssl
+import urllib2
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EFKStackCheck(OpenShiftCheck):
+    """
+        This is a check for the entire EFK stack shipped with OCP
+        https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py
+    """
+
+    name = "efk_stack_check"
+    tags = ["elasticsearch", "efkstack"]
+
+    namespace = "logging"
+
+    es_pods = []
+    fluentd_pods = []
+
+    def __init__(self, module_executor):
+        super(EFKStackCheck, self).__init__(module_executor)
+
+        es_cert_prefix = '/etc/elasticsearch/secret/admin-'
+        self.es_curl = "curl -s --cert {}cert --key {}key --cacert {}ca -XGET ".format(es_cert_prefix, es_cert_prefix, es_cert_prefix)
+
+    def run(self, tmp, task_vars):
+        self.get_pods(task_vars)
+
+        status = dict()
+        status['elasticsearch'] = self.check_elasticsearch(task_vars)
+        status['fluentd'] = self.check_fluentd(task_vars)
+        status['kibana'] = self.check_kibana(task_vars)
+
+        if not status.get("elasticsearch", {}).get("pods", None):
+            return {"failed": True, "msg": "Unable to find any elasticsearch pods. Is elasticsearch installed on the host?"}
+
+        if not status.get("kibana"):
+            return {"failed": True, "msg": "Unable to access the Kibana logging url. No routes found."}
+        elif not status.get("kibana", {}).get("site_up"):
+            return {"failed": True, "msg": "Unable to access Kibana logging url. Invalid server response."}
+
+        return {"changed": False, "msg": status}
+
+    def oc(self, cmd_str, task_vars):
+        """ Execute an 'oc' command in the remote host """
+        config_base = get_var(task_vars, "openshift", "common", "config_base")
+        args = {
+            "namespace": self.namespace,
+            "config_file": os.path.join(config_base, "master", "admin.kubeconfig"),
+            "cmd": cmd_str,
+        }
+
+        result = self.module_executor("ocutil", args, task_vars)
+        return result.get("ansible_facts", {}).get("result", "")
```

Comment:
Will update, since per [your suggestion here](https://github.com/openshift/openshift-ansible/pull/3643#discussion_r106513615), we can read the `results` key directly from the module_executor result